linalg/hypre.hpp

```diff
@@ -570,6 +570,14 @@ HypreParMatrix * RAP(const HypreParMatrix *A, const HypreParMatrix *P);
 HypreParMatrix * RAP(const HypreParMatrix * Rt, const HypreParMatrix *A,
                      const HypreParMatrix *P);
 
+/// Returns the matrix constructed from blocks.
+/** It is assumed that all block matrices use the same communicator, and the
```

Comment:
I would add a brief description of the resulting matrix's parallel data layout (ie, the basic principle is that none of the matrix entries/data move across processors, basically only rows and columns get renumbered).

Reply:
Good point, it may be unclear to the user that this does not redistribute or load balance. I added some comments.